src/chrome/chromeDebugAdapter.ts

```diff
@@ -621,7 +667,16 @@ export abstract class ChromeDebugAdapter implements IDebugAdapter {
                 // to be resolved in this loaded script, and remove the pendingBP.
                 this._pendingBreakpointsByUrl.delete(mappedUrl);
             } else {
-                resolvePendingBPs(mappedUrl);
+                // If break on load is active and we are using the regex approach, only call the resolvePendingBreakpoint function for files where we do not
+                // set break on load breakpoints. For those files, it is called from onPaused function.
+                // For the default Chrome's API approach, we don't need to call resolvePendingBPs from inside scriptParsed
+                if (this._breakOnLoadStrategy !== 'none') {
+                    if (this._breakOnLoadStrategy === 'regex' && !this._breakOnLoadHelper.stopOnEntryRequestedFileNameToBreakpointId.has(mappedUrl)) {
```

Comment:
This is a complicated check, can there be a method like `shouldResovePendingBPs` on the helper?